### DISCUSSION TITLE
useFetcher persistent after component is unmount (so accessible with useFetchers)

### DISCUSSION THREAD

**Discussion by blasterbug** [remix-run/remix#2623] [Proposals]
Hi, 

I have this edge case where I do not control when some components are unmounted because it is done by a third-party library. Essentially, I would like to be able to keep accessing fetchers/requests started by a `useFetcher` hook, even when the component calling this hook is unmounted. So the fetcher would be still accessible from `useFecthers` hook. Is it possible to add an option to the fetcher object/Form something like `cancel: false`? 

// Ben

**Comment by kiliman** [comment]
I'm not sure what your use case is, but can't you create the fetcher from a parent component and pass it to the child as a prop?

**Reply by blasterbug** [reply]
Because the fetcher is created by my component which is mounted and unmounted by a third party library and I can not change the behaviour of that third party library. Passing the fetcher from a parent component as a prop is not either an option because I do not control that either.  

And I can imagine other scenario where an component starting a request (via the `useFetcher` hook for instance) would get unmount before it completes

To give some more context, I am using AgGrid, which renders tables as Excel would. Each cell editor, my component, is mounted and unmounted when a cell is edited.

**Reply by kiliman** [reply]
The `useFetcher` hook uses `useEffect` and `useState`, so don't think they'll be able to persist when component is unmounted. If you can't control the lifecycle of your component, then you may need to add your fetchers to React context, and your component can get it from your context provider. This is more a React issue than a Remix one.

**Reply by blasterbug** [reply]
Well, the idea was that the Remix `useFetchers` hook does a very similar job already. So if I could just tell Remix `useFetcher` hook "do not cancel this fetcher", then I have access to it from `useFetchers`. 

Besides, putting the fetcher in my own context won't solve the problem that the request got cancelled when my component gets unmounted...

**Reply by hanayashiki** [reply]
You are building too great an app using remix. You should handle data by hand in that case
